services/motion/builtin/builtin.go

```diff
@@ -24,7 +24,7 @@ import (
 
 func init() {
 	registry.RegisterService(motion.Subtype, resource.DefaultModelName, registry.Service{
-		Constructor: func(ctx context.Context, r robot.Robot, c config.Service, logger golog.Logger) (interface{}, error) {
+		RobotConstructor: func(ctx context.Context, r robot.Robot, c config.Service, logger golog.Logger) (interface{}, error) {
```

Comment:
Unfortunate that some services will need to use the legacy constructor. I made a note that we should try to remove this in the corresponding ticket for components: https://viam.atlassian.net/browse/RSDK-783?focusedCommentId=12674 FYI

Reply:
Yeah agreed, I think if we rework the frame system and pull it out of robot then we can change motion service. Also another thing to note. All the default services except sensors are using legacy as well. The idea was we were worried about the default service requiring dependencies on robots that don't have the dependency and are not using the service at all. Its probably fixable but, I didnt want to over complicate the PR

Reply:
it's an effort for another day for sure - I was just lamenting